core/templates/components/ck-editor-helpers/ck-editor-4-rte.component.ts

```diff
@@ -44,21 +45,45 @@ interface RteConfig extends CKEDITOR.config {
             '</div></div>',
   styleUrls: []
 })
-export class CkEditor4RteComponent implements AfterViewInit, OnDestroy {
+export class CkEditor4RteComponent implements AfterViewInit, OnDestroy, OnInit {
   @Input() uiConfig: UiConfig;
   @Input() value;
   @Input() headersEnabled = false;
   @Output() valueChange: EventEmitter<string> = new EventEmitter();
   rteHelperService;
   ck: CKEDITOR.editor;
+  hasNetworkConnection: boolean;
+  hasInternetAccess: boolean;
+  status: string = 'ONLINE';
+  disableOnOfflineNames: string[] = [
+    'image', 'skillreview', 'svgdiagram', 'video'];
   constructor(
     private ckEditorCopyContentService: CkEditorCopyContentService,
     private contextService: ContextService,
-    private elementRef: ElementRef
+    private elementRef: ElementRef,
+    private connectionService: ConnectionService
   ) {
     this.rteHelperService = OppiaAngularRootComponent.rteHelperService;
   }
 
+  ngOnInit(): void {
+    this.connectionService.onInternetStateChange.subscribe(currentState => {
+      this.hasNetworkConnection = currentState.hasNetworkConnection;
+      this.hasInternetAccess = currentState.hasInternetAccess;
+      if (this.hasNetworkConnection && this.hasInternetAccess) {
+        if (this.status === 'OFFLINE') {
+          this.enableRTEicons();
```

Comment:
This seems a bit odd.. when offline, why are we "enabling" icons. Did you mean disable certain icons? Or perhaps the "status" refers to the previous state of connection?

Reply:
this check is for the previous state of the connection.